File: Jan_Basic_Programming/Assignments/C/31.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <stdbool.h>
// Anagrams
char *sortedStr(char str[], int len)
{
    char *sorted = (char *)malloc((len + 1) * sizeof(char));
    char temp;

    for (int i = 0; i < len; i++)
    {
        sorted[i] = str[i];
    }

    for (int i = 0; i < len - 1; i++)
    {
        for (int j = i + 1; j < len; j++)
        {
            if (sorted[i] > sorted[j])
            {
                temp = sorted[i];
                sorted[i] = sorted[j];
                sorted[j] = temp;
            }
        }
    }

    sorted[len] = '\0';
    return sorted;
}

int main()
{
    char str1[30], str2[30];
    bool flag = false;

    printf("Enter the First String : ");
    scanf(" %s", &str1);
    printf("Enter the second String : ");
    scanf(" %s", &str2);

    if (strlen(str1) != strlen(str2))
    {
        printf("Given String are not Anagram.");
    }
    else
    {
        char *sortedStr1 = sortedStr(str1, strlen(str1));
        char *sortedStr2 = sortedStr(str2, strlen(str2));

        for (int i = 0; i < strlen(str1); i++)
        {
            if (sortedStr1[i] != sortedStr2[i])
            {
                flag = true;
                break;
            }
        }

        if (flag)
        {
            printf("Given String are not Anagram.");
        }
        else
        {
            printf("Given String are Anagram.");
        }
    }

    return 0;
}

File: Mar_PHP/Assignments/BasicPrograms/31.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Anagram Checker</title>
</head>
<body>
    <h2>Anagram Checker</h2>
    <form method="post">
        Enter first string: <input type="text" name="str1" required>
        <br><br>
        Enter second string: <input type="text" name="str2" required>
        <br><br>
        <input type="submit" name="submit" value="Check Anagram">
    </form>

    <?php
    function isAnagram($str1, $str2) {
        
        $str1 = strtolower(str_replace(' ', '', $str1));
        $str2 = strtolower(str_replace(' ', '', $str2));

        
        $sorted_str1 = str_split($str1);
        $sorted_str2 = str_split($str2);
        sort($sorted_str1);
        sort($sorted_str2);

        
        return $sorted_str1 == $sorted_str2;
    }

    if (isset($_POST['submit'])) {
        $str1 = $_POST['str1'];
        $str2 = $_POST['str2'];

        if (isAnagram($str1, $str2)) {
            echo "<p>'$str1' and '$str2' are anagrams.</p>";
        } else {
            echo "<p>'$str1' and '$str2' are not anagrams.</p>";
        }
    }
    ?>
</body>
</html>
